<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"></ul></div>

In [1]:
# import packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# read csv file
df = pd.read_csv('2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv')

In [3]:
# check the data frame 
df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11

In [4]:
# check overall variables type and counts
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   order_id        5000 non-null   int64 
 1   shop_id         5000 non-null   int64 
 2   user_id         5000 non-null   int64 
 3   order_amount    5000 non-null   int64 
 4   total_items     5000 non-null   int64 
 5   payment_method  5000 non-null   object
 6   created_at      5000 non-null   object
dtypes: int64(5), object(2)
memory usage: 273.6+ KB


In [5]:
# check distribution of each numeric columns
df.describe()

order_id      shop_id      user_id   order_amount  total_items
count  5000.000000  5000.000000  5000.000000    5000.000000   5000.00000
mean   2500.500000    50.078800   849.092400    3145.128000      8.78720
std    1443.520003    29.006118    87.798982   41282.539349    116.32032
min       1.000000     1.000000   607.000000      90.000000      1.00000
25%    1250.750000    24.000000   775.000000     163.000000      1.00000
50%    2500.500000    50.000000   849.000000     284.000000      2.00000
75%    3750.250000    75.000000   925.000000     390.000000      3.00000
max    5000.000000   100.000000   999.000000  704000.000000   2000.00000

**Note: as we can see, the max of order_amount and total_items are extremely high, which might be the outliers of the dataset and introduce screw of the distribution.**

In [6]:
# take a closer look of the total_items value counts
df['total_items'].value_counts()

2       1832
1       1830
3        941
4        293
5         77
2000      17
6          9
8          1
Name: total_items, dtype: int64

**Note: there are 17 orders have 2000 items in each order, which does not make too much sense in real case, there might be some technical issues when acquiring these data.Therefore, I am going to drop these outliers because it only 17/5000=0.34% of the total dataset.**

In [7]:
# drop the outliers
df_clean = df.drop(df[df['total_items']==2000].index, axis=0)

In [8]:
# sanity check
df_clean['total_items'].value_counts()

2    1832
1    1830
3     941
4     293
5      77
6       9
8       1
Name: total_items, dtype: int64

In [9]:
# calculate the AOV
df_clean['order_amount'].mean()

754.0919125025085

**ANSWER: The AOV of orders of shoes in 30 days window is $754.**

In [27]:
# one step futher, I also list the AOV for each store which gives us more deatails insights
df_store_aov = pd.DataFrame(df.groupby('shop_id')['order_amount'].mean()).reset_index()
df_store_aov

shop_id  order_amount
0         1    308.818182
1         2    174.327273
2         3    305.250000
3         4    258.509804
4         5    290.311111
..      ...           ...
95       96    330.000000
96       97    324.000000
97       98    245.362069
98       99    339.444444
99      100    213.675000

[100 rows x 2 columns]